pkg/plugin/processor/builtin/convertField_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+	"testing"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/matryer/is"
+)
+
+func TestConvertField_Process(t *testing.T) {
+	proc := convertField{}
+	ctx := context.Background()
+	var err error
+	testCases := []struct {
+		name   string
+		field  string
+		typ    string
+		record opencdc.Record
+		want   sdk.SingleRecord
+	}{
+		{
```

Comment:
Nitpick: maybe have a few negative test cases, e.g. string to int, where strings is not a valid number, or converting a very big float into an int and so on.